src/main/java/com/microsoft/azure/vmagent/util/TokenCache.java

```diff
@@ -103,14 +93,24 @@ public static ApplicationTokenCredentials get(final AzureCredentials.ServicePrin
     }
 
     public Azure getAzureClient() throws AzureCloudException {
+        if (client != null) {
+            return client;
+        }
         try {
-            return Azure
-                    .configure()
-                    .withInterceptor(new AzureVMAgentPlugin.AzureTelemetryInterceptor())
-                    .withLogLevel(Constants.DEFAULT_AZURE_SDK_LOGGING_LEVEL)
-                    .withUserAgent(getUserAgent())
-                    .authenticate(get(credentials))
-                    .withSubscription(credentials.getSubscriptionId());
+            synchronized (this) {
+                if (client != null) {
+                    return client;
+                } else {
+                    client = Azure
+                            .configure()
+                            .withInterceptor(new AzureVMAgentPlugin.AzureTelemetryInterceptor())
+                            .withLogLevel(Constants.DEFAULT_AZURE_SDK_LOGGING_LEVEL)
+                            .withUserAgent(getUserAgent())
+                            .authenticate(get(credentials))
```

Comment:
What does the `authenticate` method actually perform? I think it's better to go deeper. If `authenticate` talks with azure to get/refresh something like `AccessToken` which will expire in certain time, it could be a problem with the `client`.